src/storage.cpp

```diff
@@ -907,17 +917,17 @@ namespace libtorrent
 				// the resume data indicates we're a seed, but this file has
 				// the wrong size. Reject the resume data
 				ec.ec = errors::mismatching_file_size;
-				ec.file = file_index;
+				ec.file(file_index);
 				ec.operation = storage_error::check_resume;
 				return false;
 			}
 
 			// OK, this file existed, good. Now, skip all remaining pieces in
 			// this file. We're just sanity-checking whether the files exist
 			// or not.
-			peer_request const pr = fs.map_file(file_index, 0
-				, int(fs.file_size(file_index) + 1));
```

Comment:
the only output that's used from this function is the piece (ifaict). and the file size passed to it is truncated so that can't be relied on anyway. the output piece is determined by the input file and offset. the offset does support large files.